### PR TITLE
Add NPM Publishing workflow

### PR DESCRIPTION
**Pull request by EvanBoyle**
also removes global state for tests. I created the NPM org `@cortexclick` and the package will be published under `@cortexclick/cortex`. 

Will run the first release after this is merged, there is a chance there will be some trial/error getting things completely working e2e, but unfortunately can't do much testing without merging this workflow.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "typescript-eslint": "^7.13.1",
     "vitest": "^1.6.0"
   },
-  "name": "cortex-sdk",
+  "name": "@cortexclick/cortex",
```
Comment (jmoseley) [inline package.json:16]: I'm worried this package name is confusing. Should it be like `@cortexclick/js` or `@cortexclick/sdk`? `cortex` does not really describe this package.

**Review by jmoseley — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (EvanBoyle) [reply package.json:16]: People do all sorts of differnt things with package names. I prefer to stick to something branded for the SDK. Similar to openAI and pulumi: 

- OpenAI's SDK is just called `openai` https://www.npmjs.com/package/openai
- Pulumi's SDK is `@pulumi/pulumi` https://www.npmjs.com/package/@pulumi/pulumi

But others do just use sdk: 

- Anthropic: https://www.npmjs.com/package/@anthropic-ai/sdk

Comment (EvanBoyle) [reply package.json:16]: And `cortex` is the key noun for our offering. A `cortex` is a customizable AI agent for marketing/sales automation that let's you operate over your enterprise data. 

We will have some work to do one the marketing and brand awareness side, but I think we should lean into it and not away from it.

Comment (chaosrealm) [reply package.json:16]: Why not `cortex-sdk`?

**Review by chaosrealm — COMMENTED** [review]
lgtm

**Pull request merged**